### PR TITLE
Publishing Microsoft.Azure.Monitor.dll for the first time before build

### PR DESCRIPTION
**Pull request by gucalder**
…branch

## Description
This PR is to **publish the Microsoft.Azure.Monitor.dll for the first time** before build (and start the deprecation of Microsoft.Azure.Insights.dll).

This release **introduces the breaking changes announced months ago**. It also modifies some the APIs (non breaking changes), fixes some bugs, improves documentation and is closer to full compliance with guidelines (i.e. the check list for the Swagger specs.)

This PR is also merging AutoRest branch changes to issue a pull request from vs17Dev as requested during PR: https://github.com/Azure/azure-sdk-for-net/pull/3075.

_The code was generated from the latest Swagger spec_. (See PR https://github.com/Azure/azure-rest-api-specs/pull/1094) 
No more changes are scheduled to be included for build.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as a link to the swagger spec, used to generate the code.
- [x] The `project.json` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by shahabhijeet** [comment]
@gucalder this branch is still not opened and accepting PR. An announcement will be sent once the branch is opened. Please reopen PR once the branch is open.

**Comment by gucalder** [comment]
I just reopened this PR since I was told vs17Dev is already accepting PRs.

**Comment by gucalder** [comment]
The errors are due to nuget issues from other teams' folders.

**Comment by gucalder** [comment]
The checks never came back and there is no link to them. Do not know what happened with them.

**Comment by gucalder** [comment]
This one does not come back either, do not know why.

**Comment by shahabhijeet** [comment]
@azuresdkci test this please

File: src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj
Comment (shahabhijeet) [inline src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj]: @gucalder any reason the assembly name is changing?
I see more reason to cause confusion as the package name differs from assembly name.

File: src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj
Comment (shahabhijeet) [inline src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj]: @gucalder package name and assembly name should match, unless you have a reason to separate them out.

File: src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj
Comment (shahabhijeet) [inline src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj]: @gucalder all nuget package related properties that have been added are not needed, they are controlled from a central location to have consistent nuget property values across all azure sdks

File: src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj
Comment (shahabhijeet) [inline src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj]: @gucalder any reason this package is targeting all these frameworks. We are only targeting net452 and netstandard 1.4, others should be removed.

File: src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj
Comment (shahabhijeet) [inline src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj]: @gucalder why root namespace differs from any of the namespace that is either available as part of package id or assembly names.
Let's make all these things consistent across this entire SDK

File: src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj
Comment (shahabhijeet) [inline src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj]: @gucalder please remove commented code

File: src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj
Comment (shahabhijeet) [inline src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj]: @gucalder why the version remains the same? and why is it starting from 0.16.0 as this is sort of v1 for this sdk?

File: src/SDKs/Monitor/Management.Monitor/Generated/Monitor/TenantActivityLogsOperations.cs
Comment (shahabhijeet) [inline src/SDKs/Monitor/Management.Monitor/Generated/Monitor/TenantActivityLogsOperations.cs]: @gucalder if this is a new SDK, the root namespace should match the new SDK.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj
Comment (gucalder) [inline src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj]: This is a rename of the previous SDK dll, Microsoft.Azure.Insights.dll, with one breaking change for Metrics/MetricDefinitions API. The last version for Insights was 0.15.X.X-preview, so I changed the version to 16 to emphasize that the two major changes: name and Metrics API. All the other APIs in the dll remain unchanged.
This is the first time I will publish our SDK with this name, so I kept the version at 16.

File: src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj
Comment (gucalder) [inline src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj]: Our previous assembly name (and Swagger spec) was confused with AppInsights and perhaps other assemblies. So after some requests from other teams we changed the name.
Also, our team is called Monitor after the last re-org, before we were the Insights team.

File: src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj
Comment (gucalder) [inline src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj]: This is a rename of the previous SDK dll, Microsoft.Azure.Insights.dll, with one breaking change for Metrics/MetricDefinitions API. The last version for Insights was 0.15.0-preview, so I changed the version to 16 to emphasize that the two major changes: name and Metrics API. All the other APIs in the dll remain unchanged.
This is the first time I will publish our SDK with this name, so I kept the version at 16.

File: src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj
Comment (gucalder) [inline src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj]: Other teams use the "Management" before their team name, but I have no strong opinion here.
I am using "Microsoft.Azure.Monitor" for both.

File: src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj
Comment (gucalder) [inline src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj]: Nice, removing all that.

File: src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj
Comment (gucalder) [inline src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj]: Romoved.

File: src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj
Comment (gucalder) [inline src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj]: Ok, changing the namespace for consistency.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
@gucalder this PR will not make it, unless the naming conventions are followed for what we have been using for all management plane nuget packages.
Please use Microsoft.Azure.Management.Monitor.
THis also applies to the namespace as well as assemblyname.

**Comment by gucalder** [comment]
Changing the class namespace and assembly name for compliance.

**Comment by shahabhijeet** [comment]
@vishrutshah could you also approve this PR in terms of API version.

File: src/SDKs/Monitor/Management.Monitor/Generated/Management/Monitor/DiagnosticSettingsOperations.cs
Comment (vishrutshah) [inline src/SDKs/Monitor/Management.Monitor/Generated/Management/Monitor/DiagnosticSettingsOperations.cs]: Based on my conversation with @vcanaa this is wrong api-version coming from https://github.com/Azure/azure-rest-api-specs/pull/1094/files#diff-be41c214a13a2cfc2d7397b06fc23f1dR5 This supposed to be `2017-05-01-preview`

@gucalder Do we have a tests for this operation?

**Review by vishrutshah — CHANGES_REQUESTED** [review]
I'd not merge this changes if it is going to require the swagger changes.

File: src/SDKs/Monitor/Management.Monitor/Generated/Management/Monitor/DiagnosticSettingsOperations.cs
Comment (gucalder) [inline src/SDKs/Monitor/Management.Monitor/Generated/Management/Monitor/DiagnosticSettingsOperations.cs]: Vinicius is changing the Swagger spec to remove the new diagnostic settings API.
I got his changes locally and generated the code, I am updating this PR for the sake of time.

File: src/SDKs/Monitor/Management.Monitor/Generated/Management/Monitor/Models/MetricSettings.cs
Comment (vishrutshah) [inline src/SDKs/Monitor/Management.Monitor/Generated/Management/Monitor/Models/MetricSettings.cs]: Is this change intentional? I guess it has nothing to do with removal of diagnosticsettings api, that is why just wanted to confirm.

**Review by vishrutshah — CHANGES_REQUESTED** [review]

File: src/SDKs/Monitor/Management.Monitor/Generated/Management/Monitor/Models/MetricSettings.cs
Comment (gucalder) [inline src/SDKs/Monitor/Management.Monitor/Generated/Management/Monitor/Models/MetricSettings.cs]: The change is consequence of not using the new API. That Category is new and optional in the new DiagnosticSetting API. Since we are going back to the old API, that Category will not be there.

**Comment by vishrutshah** [comment]
@shahabhijeet Changes from my swagger perspective LGTM. I am approving it from that side 👍

**Review by vishrutshah — APPROVED** [review]

File: src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj
Comment (shahabhijeet) [inline src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj]: @gucalder why default namespace is any different from the assembly name or nuget package name.
Is this just an oversight or is this intentional?

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj
Comment (gucalder) [inline src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj]: Oversight for sure. Fixing it...

**Comment by gucalder** [comment]
The reasons for the failure are not clear to me. But it seems to be that the build failed completely.

**Comment by shahabhijeet** [comment]
@azuresdkci test this please

**Comment by gucalder** [comment]
DataLake test failing.

**Comment by shahabhijeet** [comment]
@azuresdkci test this please

File: src/SDKs/Monitor/Monitor.Tests/Monitor.Tests.csproj
Comment (shahabhijeet) [inline src/SDKs/Monitor/Monitor.Tests/Monitor.Tests.csproj]: @gucalder was this a mistake? why add these files? These files will be picked up VS project system like it did in VS2017.
you don't have to do it. But even if you did, why add two different set of entries.
Please remove these entries from the project file.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

**Comment by shahabhijeet** [comment]
@gucalder once you make the minor change to the csproj file and CI passes, this PR is good to go.

**Comment by gucalder** [comment]
Done.

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**
